test/test_operators.py

```diff
@@ -0,0 +1,283 @@
+    momentum = make_obj_array([zeros+1.0 for _ in range(dim)])
+    if grad:
+        dm = make_obj_array([zeros+0.0 for _ in range(dim)])
+        de = _trig_test_func(actx, x_vec, grad=True)
+        dp = make_obj_array([np.empty(0) for _ in range(dim)])
+        dy = (
+            momentum * make_obj_array([np.empty(0) for _ in range(0)]).reshape(-1, 1)
+        )
+        for i in range(dim):
+            dp[i] = _coord_test_func(actx, x_vec, order=2, fac=(i+1), grad=True)
+        return make_conserved(dim=dim, mass=dm, energy=de, momentum=np.stack(dp),
+                              species_mass=dy)
+    else:
+        mass = zeros + 1.0
+        energy = _trig_test_func(actx, x_vec)
+        for i in range(dim):
+            momentum[i] = _coord_test_func(actx, x_vec, order=2, fac=(i+1))
+        return make_conserved(dim=dim, mass=mass, energy=energy, momentum=momentum)
+
+
+def central_flux_interior(actx, discr, int_tpair):
+    """Compute a central flux for interior faces."""
+    normal = thaw(actx, discr.normal(int_tpair.dd))
+    flux_weak = central_scalar_flux(int_tpair, normal)
+    return discr.project(int_tpair.dd, "all_faces", flux_weak)
+
+
+def central_flux_boundary(actx, discr, soln_func, btag):
+    """Compute a central flux for boundary faces."""
+    boundary_discr = discr.discr_from_dd(btag)
+    bnd_nodes = thaw(actx, boundary_discr.nodes())
+    soln_bnd = soln_func(actx=actx, x_vec=bnd_nodes)
+    bnd_nhat = thaw(actx, discr.normal(btag))
+    from grudge.trace_pair import TracePair
+    bnd_tpair = TracePair(btag, interior=soln_bnd, exterior=soln_bnd)
+    flux_weak = central_scalar_flux(bnd_tpair, bnd_nhat)
+    return discr.project(bnd_tpair.dd, "all_faces", flux_weak)
+
+
+@pytest.mark.parametrize("dim", [1, 2, 3])
+@pytest.mark.parametrize("order", [1, 2, 3])
+@pytest.mark.parametrize("test_func", [partial(_coord_test_func, order=0),
+                                       partial(_coord_test_func, order=1),
+                                       partial(_coord_test_func, order=1, fac=2),
+                                       partial(_coord_test_func, order=2),
+                                       _trig_test_func,
+                                       _cv_test_func])
+def test_grad_operator(actx_factory, dim, order, test_func):
+    """Test the gradient operator for sanity.
+
+    Check whether we get the right answers for gradients of analytic functions with
+    some simple input fields and states:
+    - constant
+    - multilinear funcs
+    - quadratic funcs
+    - trig funcs
+    - :class:`~mirgecom.fluid.ConservedVars` composed of funcs from above
+    """
+    actx = actx_factory()
+
+    tol = 1e-10 if dim < 3 else 1e-9
+    from pytools.convergence import EOCRecorder
+    eoc = EOCRecorder()
+
+    for nfac in [1, 2, 4]:
+
+        npts_axis = (nfac*4,)*dim
+        box_ll = (0,)*dim
+        box_ur = (1,)*dim
+        mesh = _get_box_mesh(dim, a=box_ll, b=box_ur, n=npts_axis)
```

Comment:
Ok, I changed these to use the actual h_max from the grid. FWIW, when I was looking over your operator tests to get a handle on the symbolic stuff - it looked like you were doing the exact same scaling for your grids as I was doing here.

Reply:
Oh, I was just suggesting here to use `nels_axis` instead of `npts_axis` (and also `nelement_per_axis` inside `_get_box_mesh`), so that it matches up with the spacing specified in `eoc.add_data_point(...)`. (If scaling by points instead of elements, I think it would need to be `1/(n-1)` instead of `1/n` in order to be consistent.)

Reply:
Ah, duh, ok.  As usual I went down a rabbit hole with this. Think it's ok now that it uses `h_max` routine?

Reply:
Not sure. @thomasgibson?

Reply:
After discussing with @thomasgibson, I think we agree that the current EOC tests should be made consistent with other wrt how they scale the grids and what grid/spacing metric is used in the EOC.  (#486)